### PR TITLE
Add Resize-19: half_pixel_symmetric

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

### Description

This PR extends Resize with a new coordinate transformation mode `half_pixel_symmetric`.

The new coordinate transformation mode is similar to the default `half_pixel` but it varies in the way the implicit ROI is placed when the scale (explicit or calculated) does not result in a whole number of pixels (fractional target output size). In such case, the former `half_pixel` places the ROI at the beginning of the input space. This makes the original `half_pixel` transformation not flip invariant. The new `half_pixel_symmetric` centers the resulting ROI so that it occupies the same relative position in the output canvas.

For example
```
data = [100, 200]
scale = 1.3
resize(data, scale, 'half_pixel') -> [100.         165.38461962]
resize(data, scale, 'half_pixel_symmetric') -> [111.53846013 188.46153987]
```
We can observe also the flip invariant property in the new coordinate transformation:
```
resize(data[::-1], scale, 'half_pixel') -> [200.         134.61538038]
resize(data, scale, 'half_pixel')[::-1] -> [165.38461962 100.        ]

resize(data[::-1], scale, 'half_pixel_symmetric') -> [188.46153987 111.53846013]
resize(data, scale, 'half_pixel_symmetric')[::-1] -> [188.46153987 111.53846013]
```

### Motivation and Context

- Flip invariant property is crucial in Deep Learning applications, especially in detection tasks, where the location of the targets needs to be precise. 
- Other libraries, such as NVIDIA DALI, provide a flip invariant implementation of Resize.

### CODE REVIEW

File: docs/Changelog.md [hunk 1]
```diff
@@ -22545,6 +22545,139 @@ This version of the operator has been available since version 18 of the default
...
+ and `output_width_int` the effective output width as an integer.
+
+if coordinate_transformation_mode is `"half_pixel"`,
+```
+x_original = (x_resized + 0.5) / scale - 0.5
+```
+
+if coordinate_transformation_mode is `"half_pixel_symmetric"`,
+```
+adjustment = output_width_int / output_width;
```
Comment (xadupre) [inline docs/Changelog.md:22589]: Final `;` is intended?

File: onnx/reference/ops/op_resize.py [hunk 2]
```diff
@@ -72,7 +72,7 @@ def _nearest_coeffs(ratio: float, mode: str = "round_prefer_floor") -> np.ndarra
     raise ValueError(f"Unexpected value {mode!r}.")
 
 
-def _cubic_coeffs(ratio: float, scale: float, A: float = -0.75) -> np.ndarray:
+def _cubic_coeffs(ratio: float, scale: float = None, A: float = -0.75) -> np.ndarray:
```
Comment (gramalingam) [inline onnx/reference/ops/op_resize.py:75]: nit: `typing.Optional[float]` would be a better annotation for `scale`. Same below.

**Review by gramalingam — APPROVED** [review]

**Comment by gramalingam** [comment]
@jantonguirao : can you resolve the conflicts? Thanks!

**Comment by jantonguirao** [comment]
> @jantonguirao : can you resolve the conflicts? Thanks!

Done. Thanks

**Pull request merged**